### PR TITLE
[IIIF-166] Add framework for integration tests, add imageUploadIT

### PR DESCRIPTION
**Pull request by hardyoyo**

### CODE REVIEW

**Comment by ksclarke** [comment]
I get intermittent build failures:

```Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.106 sec <<< FAILURE! - in edu.ucla.library.bucketeer.imageUploadIT
checkThatWeCanLoadAnImage(edu.ucla.library.bucketeer.imageUploadIT)  Time elapsed: 4.509 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at edu.ucla.library.bucketeer.imageUploadIT.checkThatWeCanLoadAnImage(imageUploadIT.java:149)

Results :

Failed tests: 
  imageUploadIT.checkThatWeCanLoadAnImage:149 null
```
The file also need formatting (indentation correction, no white spaces on empty lines, etc.) to not have Checkstyle warnings.

**Comment by ksclarke** [comment]
Would you also CamelCase the class name? There isn't currently a checkstyle rule for that, but it's a Java best practice.

**Comment by ksclarke** [comment]
Moving back to in progress because Hardy has decided to do the IT on CodeBuild so that it can fully test the full process (including the kakadu step). I had inserted a cheat in for him that would make IT testing work on Travis, but he's rightly pointed out that that's not fully testing all the steps.

Also, can we remove the two assertj and rest-assured dependencies? It seems a bit heavy to add two jars so that assert statements can be written like `assertThat(myStatusCode).isEqualTo(200);` instead of like `assertEquals(200, myStatusCode);`

File: src/test/java/edu/ucla/library/bucketeer/ImageUploadIT.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                LOGGER.debug(MessageCodes.BUCKETEER_038);
+                final JsonObject jsonConfig = config.result();
+                myS3Bucket = jsonConfig.getString(Config.S3_BUCKET, DEFAULT_S3_BUCKET);
+                myS3AccessKey = jsonConfig.getString(Config.S3_ACCESS_KEY, DEFAULT_ACCESS_KEY);
+                myS3SecretKey = jsonConfig.getString(Config.S3_SECRET_KEY, DEFAULT_SECRET_KEY);
+
+                // We need to determine if we'll be able to run the S3 integration tests so we can skip if needed
+                if (jsonConfig.containsKey(Config.S3_ACCESS_KEY) && !jsonConfig.getString(Config.S3_ACCESS_KEY,
+                        DEFAULT_ACCESS_KEY).equalsIgnoreCase(DEFAULT_ACCESS_KEY)) {
+                    isExecutable = true;
```
Comment (ksclarke) [inline src/test/java/edu/ucla/library/bucketeer/ImageUploadIT.java]: Without waiting for this lambda function to complete you could end up checking isExecutable before it's been set. You need to not complete the configureRestAssured function until the lambda has been completed.

Comment (ksclarke) [reply src/test/java/edu/ucla/library/bucketeer/ImageUploadIT.java]: You can borrow the async.complete() stuff from the test where you found this config checking code and that should get it working.

**Comment by hardyoyo** [comment]
> Also, can we remove the two assertj and rest-assured dependencies? It seems a bit heavy to add two jars so that assert statements can be written like `assertThat(myStatusCode).isEqualTo(200);` instead of like `assertEquals(200, myStatusCode);`

RestAssured is no longer used, I'll finish removing it. AssertJ is doing more than what is immediately obvious. It can match strings... When I was originally writing this, I couldn't figure out how to get it to match strings. It matches integers pretty well, though, as you suggest. I'll time-box my research for doing what you ask to an hour, but if I can't figure it out, I'm sticking with AssertJ. Which does what I ask it to do. :-)

**Comment by ksclarke** [comment]
JUnit recommends (i.e., includes it as a dependency) Hamcrest matches for string contains and the like: https://junit.org/junit4/javadoc/latest/org/hamcrest/CoreMatchers.html#containsString(java.lang.String)

**Pull request merged**